src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -60,5 +65,84 @@ public ConnectionInfo(ISqlConnectionFactory factory, string ownerUri, Connection
         /// Returns true is the db connection is to a SQL db
         /// </summary>
         public bool IsAzure { get; set; }
+
+        /// <summary>
+        /// Try to get the DbConnection associated with the given connection type string. 
+        /// </summary>
+        public bool TryGetConnection(string connectionType, out DbConnection connection)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+
+            return ConnectionTypeToConnectionMap.TryGetValue(connectionType, out connection);
+        }
+
+        /// <summary>
+        /// Get a List of all DbConnection instances held by this ConnectionInfo
+        /// </summary>
+        public List<DbConnection> GetAllConnections()
+        {
+            return ConnectionTypeToConnectionMap.Values.ToList();
+        }
+
+        /// <summary>
+        /// Get a list of all connection type strings held by this ConnectionInfo
+        /// </summary>
+        /// <returns></returns>
+        public List<string> GetAllConnectionTypes()
```

Comment:
Personally, I think any method with a Get* name that just returns a value and doesn't take any parameters should be a getter-only property.